modules/common/affinity/affinity_test.go

```diff
@@ -47,13 +47,82 @@ var affinityObj = &corev1.Affinity{
 	},
 }
 
+// weightedPodAffinityTermOverride represents an Override passed to the Affinity
+// tests
+var weightedPodAffinityTermOverride = []corev1.WeightedPodAffinityTerm{
+	{
+		PodAffinityTerm: corev1.PodAffinityTerm{
+			LabelSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      "CustomKeySelector",
+						Operator: metav1.LabelSelectorOpIn,
+						Values: []string{
+							"selectorValue1",
+							"selectorValue2",
+							"selectorValue3",
+						},
+					},
+				},
+			},
+			TopologyKey: "CustomTopologyKey",
+		},
+		Weight: 80,
+	},
+}
+
 func TestDistributePods(t *testing.T) {
 
 	t.Run("Default pod distribution", func(t *testing.T) {
 		g := NewWithT(t)
+		d, _ := DistributePods("ThisSelector", []string{"selectorValue1", "selectorValue2"}, "ThisTopologyKey", nil)
+		g.Expect(d).To(BeEquivalentTo(affinityObj))
+	})
 
-		d := DistributePods("ThisSelector", []string{"selectorValue1", "selectorValue2"}, "ThisTopologyKey")
+	// Override the default AntiAffinity
+	t.Run("Pod distribution with overrides", func(t *testing.T) {
+		// The resulting affinity that should be assigned to the Pod
+		var expectedAffinity = &corev1.Affinity{
+			PodAffinity:  nil,
+			NodeAffinity: nil,
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: weightedPodAffinityTermOverride,
+			},
+		}
+		affinityOverride := &OverrideSpec{
+			PodAffinity: nil,
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: weightedPodAffinityTermOverride,
+			},
+			NodeAffinity: nil,
+		}
+		g := NewWithT(t)
+		d, _ := DistributePods("ThisSelector", []string{"selectorValue1", "selectorValue2"}, "ThisTopologyKey", affinityOverride)
+		g.Expect(d).To(BeEquivalentTo(expectedAffinity))
+	})
 
-		g.Expect(d).To(BeEquivalentTo(affinityObj))
+	// Override the Affinity but keep the default AntiAffinity
+	t.Run("Pod distribution with overrides", func(t *testing.T) {
+		// The resulting affinity that should be assigned to the Pod
+		var expectedAffinity = &corev1.Affinity{
+			// the default PodAntiAffinity defined in the DistributePods function
+			// is applied, while PodAffinity is the result of the override passed
+			// as input
+			PodAntiAffinity: affinityObj.PodAntiAffinity,
+			NodeAffinity:    nil,
+			PodAffinity: &corev1.PodAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: weightedPodAffinityTermOverride,
+			},
+		}
+		affinityOverride := &OverrideSpec{
+			PodAntiAffinity: nil,
```

Comment:
Here we see that even if you set Anti to nil, that the default behavior (of adding anti) results in it still getting added.